src/ports/items/utils.ts

```diff
@@ -21,44 +21,90 @@ export function fromItemFragment(
-    data: {
-      wearable: {
-        description: fragment.metadata.wearable.description,
-        category: fragment.metadata.wearable.category,
-        bodyShapes: fragment.searchWearableBodyShapes,
+  switch (fragment.itemType) {
+    case FragmentItemType.WEARABLE_V1:
+    case FragmentItemType.WEARABLE_V2:
+    case FragmentItemType.SMART_WEARABLE_V1: {
+      const item: Item = {
+        id: fragment.id,
+        name: fragment.metadata.wearable!.name,
+        thumbnail: fragment.image,
+        url: `/contracts/${fragment.collection.id}/items/${fragment.blockchainId}`,
+        category: NFTCategory.WEARABLE,
+        contractAddress: fragment.collection.id,
+        itemId: fragment.blockchainId,
+        beneficiary: !isAddressZero(fragment.beneficiary)
+          ? fragment.beneficiary
+          : null,
         rarity: fragment.rarity,
-        isSmart: fragment.itemType === FragmentItemType.SMART_WEARABLE_V1,
-      },
-    },
-    network,
-    chainId,
-    createdAt: +fragment.createdAt * 1000,
-    updatedAt: +fragment.updatedAt * 1000,
-    reviewedAt: +fragment.reviewedAt * 1000,
-    //@ts-ignore
-    soldAt: +fragment.soldAt * 1000,
-  }
+        price: MAX_ITEM_PRICE === fragment.price ? '0' : fragment.price,
+        available: +fragment.available,
+        isOnSale:
+          fragment.searchIsStoreMinter &&
+          +fragment.available > 0 &&
+          fragment.price !== '0',
+        creator: fragment.collection.creator,
+        data: {
+          wearable: {
+            description: fragment.metadata.wearable!.description,
+            category: fragment.metadata.wearable!.category,
+            bodyShapes: fragment.searchWearableBodyShapes!,
+            rarity: fragment.rarity,
+            isSmart: fragment.itemType === FragmentItemType.SMART_WEARABLE_V1,
+          },
+        },
+        network,
+        chainId,
+        createdAt: +fragment.createdAt * 1000,
+        updatedAt: +fragment.updatedAt * 1000,
+        reviewedAt: +fragment.reviewedAt * 1000,
+        soldAt: +fragment.soldAt * 1000,
+      }
+
+      return item
+    }
+    case FragmentItemType.EMOTE_V1: {
+      const item: Item = {
+        id: fragment.id,
+        name: fragment.metadata.emote!.name,
+        thumbnail: fragment.image,
+        url: `/contracts/${fragment.collection.id}/items/${fragment.blockchainId}`,
+        category: NFTCategory.EMOTE,
+        contractAddress: fragment.collection.id,
+        itemId: fragment.blockchainId,
+        beneficiary: !isAddressZero(fragment.beneficiary)
+          ? fragment.beneficiary
+          : null,
+        rarity: fragment.rarity,
+        price: MAX_ITEM_PRICE === fragment.price ? '0' : fragment.price,
+        available: +fragment.available,
+        isOnSale:
+          fragment.searchIsStoreMinter &&
+          +fragment.available > 0 &&
+          fragment.price !== '0',
+        creator: fragment.collection.creator,
+        data: {
+          emote: {
```

Comment:
Same as previous comment, can this be abstracted so only the different params need to be provided to avoid repeating the boilerplate?

Reply:
done 👍